openmdao/jacobians/tests/test_jacobian.py

```diff
@@ -778,5 +756,56 @@ def compute_partials(self, inputs, partials):
         with assertRaisesRegex(self, KeyError, msg.format('y', 'x')):
             J = prob.compute_totals(of=['comp.y'], wrt=['p.x'])
 
+    def test_one_src_2_tgts_with_src_indices_error(self):
+        size = 10
+        prob = Problem()
+        indeps = prob.model.add_subsystem('indeps', IndepVarComp('x', np.ones(size)))
+
+        G1 = prob.model.add_subsystem('G1', Group())
+        G1.add_subsystem('C1', ExecComp('z=2.0*y+3.0*x', x=np.zeros(size//2), y=np.zeros(size//2),
+                                        z=np.zeros(size//2)))
+
+        prob.model.jacobian = DenseJacobian()
+
+        prob.model.add_objective('G1.C1.z')
+        prob.model.add_design_var('indeps.x')
+
+        prob.model.connect('indeps.x', 'G1.C1.x', src_indices=[0,1,2,3,4])
+        prob.model.connect('indeps.x', 'G1.C1.y', src_indices=[5,6,7,8,9])
+
+        prob.setup(check=False)
+
+        with self.assertRaises(Exception) as context:
+            prob.final_setup()
+        self.assertEqual(str(context.exception),
+                         "Jacobian assembly failure.  Output 'indeps.x' is connected to multiple "
+                         "inputs ['G1.C1.x', 'G1.C1.y'] on the same component using src_indices. "
+                         "To correct this issue, replace the multiple inputs with a single input "
+                         "and handle splitting the entries inside the component.")
+
+    def test_one_src_2_tgts_csc_error(self):
+        size = 10
+        prob = Problem()
+        indeps = prob.model.add_subsystem('indeps', IndepVarComp('x', np.ones(size)))
+
+        G1 = prob.model.add_subsystem('G1', Group())
+        G1.add_subsystem('C1', ExecComp('z=2.0*y+3.0*x', x=np.zeros(size), y=np.zeros(size),
+                                        z=np.zeros(size)))
+
+        prob.model.jacobian = CSCJacobian()
+
+        prob.model.add_objective('G1.C1.z')
+        prob.model.add_design_var('indeps.x')
+
+        prob.model.connect('indeps.x', 'G1.C1.x')
+        prob.model.connect('indeps.x', 'G1.C1.y')
+
+        prob.setup(check=False)
+        prob.run_model()
+
+        # before ths fix, this would raise an exception due to an indexing error if CSCJacobian
+        # was used.
```

Comment:
there's a typo in the comment, but I wonder if referencing the "this fix" is useful outside the context of this PR. 
maybe the comment should just explain the config that is verified to be valid by this test.

Reply:
I reworked this test.